include/robot_interfaces/sensor_backend.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+
+#include <real_time_tools/process_manager.hpp>
+#include <real_time_tools/thread.hpp>
+#include <real_time_tools/timer.hpp>
+
+#include <robot_interfaces/opencv_driver.hpp>
+#include <robot_interfaces/sensor_data.hpp>
+
+namespace robot_interfaces
+{
+/**
+ * @brief Communication link between SensorData and SensorDriver.
+ *
+ * At each instant, it checks if the camera can be accessed and
+ * the video capture is running, if yes, then grabs an image and
+ * stores it along with the timestamp at which it was grabbed.
+ *
+ * @tparam OpenCVObservation
```

Comment:
The template argument should just be `Observation`.  `OpenCVObservation` would be a possible value to pass.